frontend/packages/kubevirt-plugin/src/k8s/wrapper/common/k8s-resource-object-with-type-property-wrapper.ts

```diff
@@ -0,0 +1,52 @@
+/* eslint-disable lines-between-class-members */
+import { getName, hasLabel, getLabels } from '@console/shared/src';
+import { apiVersionForModel, K8sKind, K8sResourceKind } from '@console/internal/module/k8s';
+import { K8sResourceKindMethods } from '../types/types';
+import { ObjectWithTypePropertyWrapper } from './object-with-type-property-wrapper';
+import { ObjectEnum } from '../../../constants';
+
+export class K8sResourceObjectWithTypePropertyWrapper<
+  RESOURCE extends K8sResourceKind,
+  TYPE extends ObjectEnum<string>,
+  COMBINED_TYPE_DATA,
+  SELF extends K8sResourceObjectWithTypePropertyWrapper<RESOURCE, TYPE, COMBINED_TYPE_DATA, SELF>
```

Comment:
IIUC, this type param is only needed to achieve proper `this` polymorphic typing (`return this` should infer to sub-class type, not base-class type).

Also @suomiy please be aware that `SELF` referencing itself creates a recursion within the type param signature. TypeScript can handle this, but sometimes such recursions can lead to problems and should be avoided if possible.

Reply:
yes, we should remove this in the future